### DISCUSSION TITLE
SPI initialization failed

### DISCUSSION THREAD

**Discussion by yiming564** [Bodmer/TFT_eSPI#2107] [General]
I'm trying to use ESP32-WROVER-E**(8MB)** to drive an SPI Screen (ST7789).
However, it failed to initialize VSPI. SCLK has no waveform
I tested SPICLASS and found that the problem still exists. (Maybe it is the problem of arduino framework?)
After changed the chip to a 4MB one, it seemed to work smoothly.
Is there a solution to initialize VSPI on the 8MB one?

**Comment by Bodmer** [comment]
There does seem to be a problem with the E variant. As reported in #2071 for a parallel setup. Maybe this is related to support in the Arduino environment.

Do other lines like CS and DC toggle?

Have you tried using the [HSPI port](https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup27_RPi_ST7796_ESP32.h#L99)?

**Reply by yiming564** [reply]
Yes, I have. However, neither HSPI nor VSPI works.
These are my hardware connections:
![image](https://user-images.githubusercontent.com/78789251/197553540-7913f0b4-6254-4571-9d7f-916e1cb02b10.png)
![image](https://user-images.githubusercontent.com/78789251/197555578-6b0b6335-accc-4db4-92c7-f338829576bc.png)
Display board:
![image](https://user-images.githubusercontent.com/78789251/197555657-056bcc27-a582-416c-be2b-7eb7f971fcab.png)
partitions.csv:
```
# ESP-IDF Partition Table
# Name,		Type,	SubType,	Offset,		Size,	Flags
nvs,		data,	nvs,		0x9000,		0x4000,
otadata,	data,	ota,		0xd000,		0x2000,
phy_init,	data,	phy,		0xf000,		0x1000,
factory,	app,	factory,	0x10000,	2M,
spiffs,		data,	spiffs,		0x210000,	1M,
ota_0,		app,	ota_0,		0x310000,	2M,
ota_1,		app,	ota_1,		0x510000,	2M,
```
User_Setup.h:
```
#define ST7789_DRIVER
#define TFT_WIDTH  240 // ST7789 240 x 240 and 240 x 320
#define TFT_HEIGHT 320 // ST7789 240 x 320
#define TFT_MISO	19
#define TFT_MOSI	23
#define TFT_SCLK	18
#define TFT_CS		4
#define TFT_DC		12
#define TFT_RST 	26
#define TFT_BL		25
#define TOUCH_CS	5
#define TOUCH_IRQ	13
#define SPI_FREQUENCY  80000000
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000
```

**Reply by Bodmer** [reply]
Try avoiding the use of pin 12 as it may be stopping the boot if the display is pulling it high, see here:
https://randomnerdtutorials.com/esp32-pinout-reference-gpios/

Other than that I have run out of ideas for the root cause of the problem.
